geode-core/src/main/java/org/apache/geode/cache/client/internal/ConnectionImpl.java

```diff
@@ -54,11 +54,16 @@
  */
 public class ConnectionImpl implements Connection {
 
+
   // TODO: DEFAULT_CLIENT_FUNCTION_TIMEOUT should be private
   public static final int DEFAULT_CLIENT_FUNCTION_TIMEOUT = 0;
   private static final String CLIENT_FUNCTION_TIMEOUT_SYSTEM_PROPERTY =
       DistributionConfig.GEMFIRE_PREFIX + "CLIENT_FUNCTION_TIMEOUT";
 
+  private static final int CLIENT_FUNCTION_TIMEOUT =
```

Comment:
I recommend changing this to a `private final int clientFunctionTimeout` field that is read from the system property in the constructor instead of setting a `static final` constant. As a constant, the value is set once and sticks -- it can never change again within the life of the JVM. So, in the tests, there's actually no effect of setting/unsetting the system property or otherwise changing the value from test-to-test (one could imagine wanting to have different values for different tests).

Reply:
I agree that in this case the constructor would be appropriate given that number of connections we make are infrequent so there shouldn't be an enormous cost here. I have since come to the conclusion though that this timeout has no business on connection anyway and would like it moved to functions where it belongs.